src/private/emscripten_api/emscripten_indoor_map_entity_information_api.js

```diff
@@ -63,77 +62,109 @@ function EmscriptenIndoorMapEntityInformationApi(emscriptenApiPointer, cwrap, ru
 
-            success = _indoorEntityInformationApi_TryGetIndoorMapEntityIdBufferSize(
-                _emscriptenApiPointer,
-                indoorMapEntityModelIds[entityIndex],
-                indoorMapEntityStringIdSizeBuf.ptr
-            );
+            var indoorMapEntityFloorId = indoorMapEntityFloorIds[i];
 
-            var indoorMapEntityStringIdSize = _emscriptenMemory.consumeBufferToArray(indoorMapEntityStringIdSizeBuf)[0];
-            
-            if(!success)
-            {
-                return null;
-            }
-            
-            var indoorMapEntityStringId = _emscriptenMemory.createInt8Buffer(indoorMapEntityStringIdSize);
-            var indoorMapFloorIdBuf = _emscriptenMemory.createInt32Buffer(1);
-            var positionLatLngBuf = _emscriptenMemory.createDoubleBuffer(2);
-
-            success = _indoorEntityInformationApi_TryGetIndoorMapEntity(
-                _emscriptenApiPointer,
-                indoorMapEntityModelIds[entityIndex],
-                indoorMapEntityStringId.ptr,
-                indoorMapEntityStringIdSize,
-                indoorMapFloorIdBuf.ptr,
-                positionLatLngBuf.ptr
-            );
-
-            var indoorMapEntityId = _emscriptenMemory.consumeUtf8BufferToString(indoorMapEntityStringId);
-            var indoorMapEntityFloorIdArray = _emscriptenMemory.consumeBufferToArray(indoorMapFloorIdBuf);
-            var positionLatLngArray = _emscriptenMemory.consumeBufferToArray(positionLatLngBuf);
-
-            if(!success)
-            {
-                return null;
-            }
+            var posLat = positionLatLng[2*i];
+            var posLng = positionLatLng[2*i + 1];
+            var position = L.latLng(posLat, posLng);
 
-            var indoorMapEntityFloorId = indoorMapEntityFloorIdArray[0];
+            var contourCount = contoursPerPolygon[i];
+            var polygonPoints = [];
 
-            var lat = positionLatLngArray[0];
-            var lng = positionLatLngArray[1];
-            var position = L.latLng(lat, lng);
+            var  latLngsPerContourEndIndex = latLngsPerContourHead + contourCount;
+            for (var latLngsPerContourIndex = latLngsPerContourHead; latLngsPerContourIndex < latLngsPerContourEndIndex;  latLngsPerContourIndex++)
+            {
+                var contourPoints = [];
+                
+                var latLngsDegreesCount = latLngsPerContour[latLngsPerContourIndex];
+                var latLngsDegreesEndIndex = latLngsDegreesHead + latLngsDegreesCount;
+
+                for(var pointIndex = latLngsDegreesHead; pointIndex < latLngsDegreesEndIndex; pointIndex++)
+                {
+                    var lat = latLngsDegrees[(pointIndex * 2)];
+                    var lng = latLngsDegrees[(pointIndex * 2) + 1];
+                    contourPoints.push([lat, lng]);
+
+                    latLngsDegreesHead++;
+                }                
+                polygonPoints.push(contourPoints);
```

Comment:
Is this definitely the representation of the polygon that we want? Elsewhere we construct L.latLngs, though accept arrays of interleaved [lat,lng] as options input - see eg https://github.com/wrld3d/wrld.js/blob/master/src/public/heatmap.js#L107